### PR TITLE
Merkle spec draft

### PR DESCRIPTION
**Pull request by ronnno**

### CODE REVIEW

File: behaviors/services/trie.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ 
+## Trie Key limitations
+### Uniform Key Length 
+All trie keys are of a uniform length. This means branch nodes and value nodes are mutually exclusive as there may not be two keys where one is contained in the other.
+   * Simplify trie structure by forcing values to be only in leaf nodes
+   * This means clients interpreting proofs must be aware of the uniform key length and enforce this knowledge when evaluating proofs
+   * This requirement depends on our state addressing scheme. See state storage requirements
+## Node types and structure 
+### Common properties
+All nodes have a prefix part. A Prefix part is the current node’s contribution to the key prefix represented by a path in the trie
```
Comment (noambergIL) [inline behaviors/services/trie.md]: prefix maybe empty

File: behaviors/services/trie.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+All nodes have a prefix part. A Prefix part is the current node’s contribution to the key prefix represented by a path in the trie 
+* For child nodes (of any type - branch or leaf) if the parent node had an “Even” parity bit, we encode the prefix part with an offset of 4 bits (Even and not Odd because the parent’s branch digit reverses the parity once it is appended to the parent prefix part).
+
+### Branch nodes
+In addition to the prefix part attribute, a branch node has up to 16 children.
+Each child reference corresponds to a possible prefix extension. Prefixes are represented as hexadecimal digits in base64 encoding. This is why any possible extension of the current path's prefix may be any one of 16 
+* Parity bit always refers to the number of digits from the beginning of the prefix up to and including the current node’s prefix part. 
+
+### Leaf nodes 
+Leaf nodes represent values or hashes of values associated with the state key which is represetned by the assembled prefix parts of the entire trie path leading to the leaf node from the root of the trie.
```
Comment (noambergIL) [inline behaviors/services/trie.md]: is it allowed for a leaf have a prefix

File: behaviors/services/trie.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    only double 'digit' bytes to work with.
+
+h3 >> {type: "leaf", prefix: "ff", value "2"}
+    no padding as parent was odd.
+    notice this node differs from h4 only in the value. 
+    it's enough to give it a distinct hash code. 
+
+h4 >> {type: "leaf", prefix: "ff", value "3"}
+    no padding as parent was odd.
+    notice how this node is reused in two paths of two keys.
```
Comment (noambergIL) [inline behaviors/services/trie.md]: will be better as a image

File: behaviors/services/trie.md (hunk 1 above)
Comment (ronnno) [reply behaviors/services/trie.md]: In a node a prefix may be empty. this is not written. do you suggest we add this? this section is about keys in their entirety, not partial prefixes so I'm not sure this is the right place for it. is this what you meant?

File: behaviors/services/trie.md (hunk 2 above)
Comment (ronnno) [reply behaviors/services/trie.md]: this is specified in the `## Common properties` section. what do you mean by this comment?

**Review by noambergIL — APPROVED** [review]

**Pull request merged**